### PR TITLE
kie-issues#90: The option to embed a Gist is always available in a Gist Workspace on KIE Sandbox

### LINKED ISSUE #90: The option to embed a Gist is always available in a Gist Workspace on KIE Sandbox
See original Jira: https://issues.redhat.com/browse/KOGITO-7562

### PR DESCRIPTION
**Pull request by klbowers**
Closes https://github.com/kiegroup/kie-issues/issues/90
Rather than "fixing" the described bug, e.g. disabling the option to embed a Gist when there are local changes, I instead followed Tiago's idea which can be seen in the original ticket https://issues.redhat.com/browse/KOGITO-7562. 
The option to embed a Gist will still always be available (when working on a gist folder), but will now show an alert when the Embed Modal is opened up with local changes.

### CODE REVIEW

File: packages/online-editor/src/editor/Toolbar/Share/EmbedModal.tsx [hunk 1]
```diff
@@ -169,6 +173,16 @@ export function EmbedModal(props: {
       title={i18n.embedModal.title}
       description={i18n.embedModal.description}
     >
+      {isGist && hasLocalChanges && (
+        <>
+          <Alert
+            isInline={true}
+            variant={"warning"}
+            title={`You have new changes to push. Embedding '${props.workspace.name}' won't show your latest changes.`}
```
Comment (yesamer) [inline packages/online-editor/src/editor/Toolbar/Share/EmbedModal.tsx]: @kbowers-ibm Can you please use `i18n` mechanism?

**Review by yesamer — COMMENTED** [review]
Thank you @kbowers-ibm, your change works well I have just a question.


https://user-images.githubusercontent.com/16005046/236158241-f1460621-e3c8-4e1f-b573-47cba631a160.mov

Shouldn't the alert disappear when selecting the `Current content` option? I guess in that case the message alert message could be misleading as the provided content is actually updated.

**Comment by yesamer** [comment]
@thiagoelg Can you please review this as well? Thank you :)

**Comment by klbowers** [comment]
Good point about the current content option! Have changed it so it's only enabled when Github gist is selected, and have changed the alert message to clarify that it only pertains to the Github gist option


https://user-images.githubusercontent.com/92726146/236174249-9fcfb5c8-e11f-47f4-aeb7-83230ad2fea8.mp4

**Review by yesamer — APPROVED** [review]
Thank you @kbowers-ibm!

**Comment by yesamer** [comment]
@thiagoelg Can you please review this PR? thanks

File: packages/online-editor/src/editor/Toolbar/EditorToolbarContextProvider.tsx [hunk 2]
```diff
@@ -184,6 +190,10 @@ export function EditorToolbarContextProvider(props: Props) {
           <EmbedModal
             workspace={props.workspace.descriptor}
             workspaceFile={props.workspaceFile}
+            gitStatusProps={{
+              workspaceDescriptor: props.workspace.descriptor,
+              workspaceGitStatusPromise: props.workspaceGitStatusPromise,
+            }}
```
Comment (thiagoelg) [inline packages/online-editor/src/editor/Toolbar/EditorToolbarContextProvider.tsx]:
```suggestion
            workspaceGitStatusPromise: props.workspaceGitStatusPromise,
```

The `workspaceDescriptor` is already available via `props.workspace`. No need to pass it again here.

**Review by thiagoelg — CHANGES_REQUESTED** [review]
Thanks for this fix! Left a single inline comment, but the PR looks good :)

File: packages/online-editor/src/editor/Toolbar/Share/EmbedModal.tsx [hunk 3]
```diff
@@ -169,6 +173,12 @@ export function EmbedModal(props: {
       title={i18n.embedModal.title}
       description={i18n.embedModal.description}
     >
+      {isGist && hasLocalChanges && contentSource === ContentSource.GIST && (
```
Comment (jomarko) [inline packages/online-editor/src/editor/Toolbar/Share/EmbedModal.tsx]: Reading the source code, to me it seems if `isGist` is `true` then also `contentSource === ContentSource.GIST` is true and vice verse. 

Can't we simplify this conditional statement?

**Review by jomarko — COMMENTED** [review]
Hi @kbowers-ibm . Thank you for a PR. Just one question from me.

File: packages/online-editor/src/editor/Toolbar/EditorToolbarContextProvider.tsx (hunk 2 above)
Comment (klbowers) [reply packages/online-editor/src/editor/Toolbar/EditorToolbarContextProvider.tsx]: Good point, thanks! Fixed :)

File: packages/online-editor/src/editor/Toolbar/Share/EmbedModal.tsx (hunk 3 above)
Comment (klbowers) [reply packages/online-editor/src/editor/Toolbar/Share/EmbedModal.tsx]: Good catch, thanks! Fixed :)

Comment (jomarko) [reply packages/online-editor/src/editor/Toolbar/Share/EmbedModal.tsx]: Thank you, assume this conversation as resolved and proceed with PR as needed. I am not going to do 'official' review.

**Comment by yesamer** [comment]
@jomarko @thiagoelg @kbowers-ibm Can I merge it?

**Comment by jomarko** [comment]
@yesamer fine for me, as stated above, not going to do some 'detailed review' from my side.

**Pull request merged**